### PR TITLE
Update to consume myst-spec examples for tests

### PR DESCRIPTION
**Pull request by fwkoch**
This PR aims to move mystjs towards myst-spec compliance. It is dependent on:
* https://github.com/executablebooks/myst-spec/pull/20
* https://github.com/executablebooks/markdown-it-docutils/pull/34

Updates:
- rather than using examples in `tests/myst/` this now uses examples published from myst-spec
- the main features are:
  - directive/role nodes wrapping parsed directives/roles
  - targets as separate node
  - legend node, following figure caption
  - contentReference renamed crossReference
  - blockBreak node

### CODE REVIEW

File: src/mdast/fromMarkdown.ts [hunk 1]
```diff
@@ -2,6 +2,13 @@ import { u } from 'unist-builder';
 import type { GenericNode, GenericParent, GenericText, Spec, Token } from './types';
 import { withoutTrailingNewline } from './utils';
 
+const UNHIDDEN_TOKENS = new Set([
```
Comment (fwkoch) [inline src/mdast/fromMarkdown.ts:5]: These tokens are marked as `hidden` in `markdown-it-docutils` but are not hidden in myst mdast

File: src/mdast/tokensToMyst.ts [hunk 2]
```diff
@@ -379,6 +381,28 @@ const defaultMdast: Record<string, Spec> = {
...
+    getAttrs(t) {
+      let opts = t.meta?.opts;
+      if (!opts || !Object.keys(opts).length) {
+        opts = undefined;
+      } else {
+        if (opts.class) opts.class = opts.class.join(' ');
+        Object.keys(opts).forEach((k) => {
+          // Handle flags, where option is simply present.
+          // This simple solution is very unlikely to be sufficient long term.
+          if (opts[k] === null) opts[k] = true;
```
Comment (fwkoch) [inline src/mdast/tokensToMyst.ts:395]: This handling of `opts` is really messy and should likely be handled elsewhere (`markdown-it-docutils`?). However, it doesn't affect the parsed content, it's only for adding options to the `directive` node, so it doesn't hugely matter...

File: src/mdast/tokensToMyst.ts [hunk 3]
```diff
@@ -456,6 +489,16 @@ function hoistSingleImagesOutofParagraphs(tree: Root) {
   });
 }
 
+function nestSingleImagesIntoParagraphs(tree: Root) {
```
Comment (fwkoch) [inline src/mdast/tokensToMyst.ts:492]: This turns
```
type: root
children:
  - type: image
    ...
```
into
```
type: root
children:
  - type: paragraph
    children:
      - type: image
        ...
```

It's not super useful, but allows for some roundtripping to make commonmark tests pass.

File: tests/myst.spec.ts [hunk 4]
```diff
@@ -12,11 +12,23 @@ type TestCase = {
...
   mdast: Root;
-  myst: string;
-  html: string;
+  myst?: string;
+  html?: string;
 };
 
-const directory = 'tests/myst';
+// Comment out to make these fail
+const SKIP_TESTS = [
```
Comment (fwkoch) [inline tests/myst.spec.ts:20]: Much of this is copied from `commonmark.spec.ts` - those tests are pretty much obsolete now.

File: src/mdast/tokensToMyst.ts (hunk 2 above)
Comment (rowanc1) [reply src/mdast/tokensToMyst.ts:395]: Yeah, there are a few bugs like this in docutils:
https://github.com/executablebooks/markdown-it-docutils/issues/33

**Review by rowanc1 — COMMENTED** [review]
Nice -- so good to see the tests coming from myst-spec. 🚀

**Pull request merged**